src/Engine/System/RenderSystem.cpp

```diff
@@ -17,27 +23,29 @@ RenderSystem::RenderSystem(std::vector<Component *> & components) :
 }
 
 void RenderSystem::update(float dt) {
-    /* Update components */
-    System::update(dt);
-
     /* Reset rendering display */
     glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
     glClearColor(0.2f, 0.3f, 0.4f, 1.f);
 
-    /* Loop through active shaders */
-    for (auto &shader : m_shaders) {
-        if (!shader.second->isEnabled()) {
-            continue;
-        }
+    if (s_camera) {
+        /* Loop through active shaders */
+        for (auto &shader : s_shaders) {
+            if (!shader.second->isEnabled()) {
+                continue;
+            }
+
+            shader.second->bind();
+            ///////////////////////////  TODO  ///////////////////////////
+            // pass a list of render components that are specific       //
+            // to this shader -- right now we are passing the entire    //
+            // list and expecting each shader to filter through         //
+            //////////////////////////////////////////////////////////////
 
-        shader.second->bind();
-        ///////////////////////////  TODO  ///////////////////////////
-        // pass a list of render components that are specific       //
-        // to this shader -- right now we are passing the entire    //
-        // list and expecting each shader to filter through         //
-        //////////////////////////////////////////////////////////////
-        shader.second->render(m_components);
-        shader.second->unbind();
+            // this reinterpret_cast business works because unique_ptr's data is
+            // guaranteed is the same as a pointer
+            shader.second->render(*s_camera, reinterpret_cast<const std::vector<Component *> &>(s_diffuseComponents));
```

Comment:
Every shader gets passed the diffuse components??

Reply:
No, this was temporary because the only render component that exists is, and thus the only components RenderSystem has right now are, diffuse. This system of passing a shader only pertinent components did and still does need a solution.